### PR TITLE
feat: environment variable substitution in INI config

### LINKED ISSUE #362: [Upstream PR#328] Substitute env variables in config
## Original PR
https://github.com/mcuadros/ofelia/pull/328

## Summary
Support environment variable substitution in INI config files.

## Decision Needed
- [ ] Port from upstream PR
- [ ] Won't implement (reason: ___)

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary

Closes #362 — Support `${VAR}` and `${VAR:-default}` environment variable substitution in INI config files.

### Use case

Users want to keep secrets (SMTP passwords, API keys) out of version-controlled config files, referencing them via environment variables instead.

```ini
[global]
smtp-host = ${SMTP_HOST:-mail.example.com}
smtp-password = ${SMTP_PASS}

[job-run "backup"]
schedule = @daily
image = ${BACKUP_IMAGE:-postgres:15}
command = pg_dump ${DB_NAME:-mydb}
```

### Behavior

| Syntax | Behavior |
|---|---|
| `${VAR}` | Replaced if defined and non-empty; kept literal if undefined (typos visible) |
| `${VAR:-default}` | Replaced if defined and non-empty; uses default if undefined or empty |
| `$VAR` (no braces) | **Not** substituted — cron expressions and shell commands stay safe |

### Changes

- `cli/envsubst.go` — `expandEnvVars()` using regex, `os.LookupEnv`
- `cli/config.go` — wired into `BuildFromFile` (per-file, before INI parse) and `BuildFromString`
- `cli/envsubst_test.go` — 24 unit tests covering all edge cases
- `cli/envsubst_integration_test.go` — 5 integration tests (BuildFromString + BuildFromFile)
- `docs/CONFIGURATION.md` — new "Environment Variable Substitution" section with Docker Compose tip
- `.github/workflows/ci.yml` — bump trivy-action 0.28.0 → v0.35.0 (upstream `setup-trivy@v0.2.1` yanked)
- CHANGELOG updated

## Test plan

- [x] 24 unit tests (happy path, edge cases, safety)
- [x] 5 integration tests (full config parse with env vars)
- [x] All existing tests pass
- [x] CI passes

### CODE REVIEW

File: cli/envsubst_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	}
+
+	job, ok := cfg.RunJobs["undef-test"]
+	if !ok {
+		t.Fatal("expected job 'undef-test' to exist")
+	}
+	if job.Image != "${NONEXISTENT_IMG_VAR}" {
+		t.Errorf("expected undefined var to stay literal, got %q", job.Image)
+	}
+}
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply cli/envsubst_integration_test.go]: Fixed in 8d2442d.
